src/main/java/com/puppycrawl/tools/checkstyle/checks/UniquePropertiesCheck.java

```diff
@@ -123,6 +120,48 @@ protected static int getLineNumber(List<String> lines, String keyName) {
     }
 
     /**
+     * Method returns line number the key is detected in the checked properties
+     * files first.
+     *
+     * @param fileText
+     *            {@link FileText} object contains the lines to process
+     * @param keyName
+     *            key name to look for
+     * @return line number of first occurrence. If no key found in properties
+     *         file, 0 is returned
+     */
+    protected static int getLineNumber(FileText fileText, String keyName) {
```

Comment:
ideally it should be private, but will need it in Tests to avoid even deeper refactoring.
I will create separate issue on UT refactoring, it could be done later.